src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -886,17 +888,29 @@ internal async Task<bool> StopTaskHubWorkerIfIdleAsync()
             {
                 if (this.isTaskHubWorkerStarted &&
                     this.knownOrchestrators.Values.Count(info => !info.IsDeregistered) == 0 &&
-                    this.knownActivities.Values.Count(info => !info.IsDeregistered) == 0)
+                    this.knownActivities.Values.Count(info => !info.IsDeregistered) == 0 &&
+                    this.knownEntities.Values.Count(info => !info.IsDeregistered) == 0)

```

Comment:
The other bug I found was that we weren't making sure all entity functions were deregistered before shutting down the task hub worker. I'm not aware of any issues this has caused anyone, but theoretically it could cause us to shut down too early.